From 3e61bd856bdd8cd4fb7d0dc776439c4d0959e9b1 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 09 Dec 2021 17:38:01 +0200
Subject: [PATCH] [SW-66044] gaudi2: pci: enable hot rst control by fw

So far FW is managing the response to pci hot reset, but the HW
doesn't know about it, so both FW and HW are reacting to the hot
reset and collision might occur.
In order to avoid that, the relevant register is updated flagging
the HW that FW takes control of the hot reset response.

Change-Id: Iec12017ee8f07d815594d5742c48f3d78b030bcb
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 7416a2a..9d5ff55 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -628,6 +628,12 @@
 	return mmCPU_TIMESTAMP_CNTCR;
 }
 
+void pci_dev_specific_setup(void)
+{
+	/* FW takes resposibility for HOT reset response */
+	regmap_writel(1, mmPCIE_WRAP_CPU_HOT_RST);
+}
+
 void pci_dev_specific_hot_rst(void)
 {
 	regmap_writel(0x10, mmPCIE_CORE_HOT_RST);
